src/prefect/engine/flow_runner.py

```diff
@@ -155,8 +155,26 @@ def initialize_run(  # type: ignore
         context.update(flow_name=self.flow.name)
         context.setdefault("scheduled_start_time", pendulum.now("utc"))
 
+        # Determine the current time, allowing our formatted dates in the context
+        # to rely on a manually set value
+        now = context.get("date")
+        if isinstance(now, str):
+            # Attempt to parse into a `DateTime` object
+            try:
+                now = pendulum.parse(now)
+            except Exception:
+                pass
+        if not isinstance(now, pendulum.DateTime):
+            if now is not None:
+                self.logger.warning(
+                    "`date` was set in the context manually but could not be parsed "
+                    "into a pendulum `DateTime` object. Additional context variables "
+                    "that rely on the current date i.e `today` and `tomorrow` will be "
+                    "based on the current time instead of the `date` context variable."
+                )
+            now = pendulum.now()
```

Comment:
For full compatibility, this should still pass in `utc` I think:

```suggestion
            now = pendulum.now('utc')
```

Reply:
Good catch. Will do